### PR TITLE
[Docs] Update the docs to refer to the 3.1.0 version

### PR DESCRIPTION
**Pull request by vkorukanti**
## Description
Few updates to to get the docs ready for 3.1.0 release.

### CODE REVIEW

**Comment by felipepessoto** [comment]
@vkorukanti I think hadoop 3.3.4, or anything above 3.3.1 does not work: https://github.com/delta-io/delta/pull/2413

**Comment by vkorukanti** [comment]
> @vkorukanti I think hadoop 3.3.4, or anything above 3.3.1 does not work: #2413

I was actually testing the S3 dynamo db  logstore yesterday. With the Hadoop version set to 3.3.4, worked fine.

```
./run-integration-tests.py --run-storage-s3-dynamodb-integration-tests --dbb-packages org.apache.hadoop:hadoop-aws:3.3.4 --dbb-conf io.delta.storage.credentials.provider=com.amazonaws.auth.profile.ProfileCredentialsProvider spark.hadoop.fs.s3a.aws.credentials.provider=com.amazonaws.auth.profile.ProfileCredentialsProvider --version 3.1.0 --maven-repo https://oss.sonatype.org/content/repositories/iodelta-1129
```

Do you have any cases where it doesn't work? We should get PR #2413 merged, but wondering if it is needed for Delta 3.1?

**Comment by felipepessoto** [comment]
If you enable fastS3AListFrom (https://github.com/delta-io/delta/blob/e3d00b69bee88274c03cb207a9665d57c1b8992e/storage/src/main/java/io/delta/storage/S3SingleDriverLogStore.java#L75) it should throw a method not found at runtime because the overload we use doesn't exist in 3.3.4.

Or if you change the build to use 3.3.4 and try to compile it should fail.

**Comment by vkorukanti** [comment]
> If you enable fastS3AListFrom
> 
> https://github.com/delta-io/delta/blob/e3d00b69bee88274c03cb207a9665d57c1b8992e/storage/src/main/java/io/delta/storage/S3SingleDriverLogStore.java#L75
> 
> it should throw a method not found at runtime because the overload we use doesn't exist in 3.3.4.
> Or if you change the build to use 3.3.4 and try to compile it should fail.
> 
> I'd say it is not a blocker for 3.1, as it is broken for a while

This seems to be broken since 3.0. However can't override the Hadoop version by specially mentioning the package version when starting the shell?

**Comment by vkorukanti** [comment]
> If you enable fastS3AListFrom
> 
> https://github.com/delta-io/delta/blob/e3d00b69bee88274c03cb207a9665d57c1b8992e/storage/src/main/java/io/delta/storage/S3SingleDriverLogStore.java#L75
> 
> it should throw a method not found at runtime because the overload we use doesn't exist in 3.3.4.
> Or if you change the build to use 3.3.4 and try to compile it should fail.
> 
> I'd say it is not a blocker for 3.1, as it is broken since 2.4

This seems to be broken since 3.0. However can't we override the Hadoop version by specially mentioning the package version 3.3.1 when starting the shell?

**Comment by vkorukanti** [comment]
In any case, will review the #2413 and get it merged soon.

File: docs/source/conf.py [hunk 1]
```diff
@@ -36,7 +36,6 @@
 project = 'Delta Lake'
 author = 'Delta Lake Authors'
 
-# OSS version
 version = '3.0.0'   # Replaced by copybara
 release = '3.0.0'
```
Comment (tdas) [inline docs/source/conf.py:40]: What is this?

Comment (tdas) [reply docs/source/conf.py:40]: shouldnt this be 3.1.0?

**Pull request merged**